ecs-cli/modules/cli/regcreds/generate_policy.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2015-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package regcreds
+
+import (
+	"encoding/json"
+
+	kmsClient "github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients/aws/kms"
+)
+
+// PolicyDocument contains the statements that make up an IAM policy
+type PolicyDocument struct {
+	Version   string
+	Statement []StatementEntry
+}
+
+// StatementEntry contains a set of actions and the resources they apply to
+type StatementEntry struct {
+	Effect   string
+	Action   []string
+	Resource []string
+}
+
+func generateSecretsPolicy(credEntries map[string]CredsOutputEntry, kmsClient kmsClient.Client) (string, error) {
+	policyStatements := make([]StatementEntry, 0, len(credEntries))
+
+	for _, entry := range credEntries {
+		keyARN := ""
+		if entry.KMSKeyID != "" {
+			validARN, err := getValidKeyARN(entry.KMSKeyID, kmsClient)
+			if err != nil {
+				return "", err
+			}
+			keyARN = validARN
+		}
+		statement := generatePolicyStatement(entry.CredentialARN, keyARN)
+		policyStatements = append(policyStatements, statement)
+	}
+
+	policyDoc := PolicyDocument{Version: "2012-10-17", Statement: policyStatements}
+	policyBytes, err := json.Marshal(&policyDoc)
+	if err != nil {
+		return "", err
+	}
+
+	return string(policyBytes), nil
+}
+
+func generatePolicyStatement(credARN, kmsKeyID string) StatementEntry {
```

Comment:
nit: I feel like the parameter should be named `kmsKeyARN`

Reply:
agreed. at this stage it should be an ARN, not an ID or alias